lib/bosh/BoshDirectorClient.js

```diff
@@ -16,27 +16,133 @@ const NotFound = errors.NotFound;
 const BadRequest = errors.BadRequest;
 
 class BoshDirectorClient extends HttpClient {
-  constructor(boshDirectorConfig) {
-    const boshDirector = boshDirectorConfig || config.director;
+  constructor() {
+    // activeConfigs - directors supporting lifecycle oprations
+    let activeConfigs = BoshDirectorClient.getActiveConfigs();
+    // directorConfig - directors supporting 'create' oprations
+    let coreConfigs = BoshDirectorClient.getCoreConfigs(activeConfigs);
+    // inactiveConfigs - directors for OutOfBand backup scenario
+    let inactiveConfigs = BoshDirectorClient.getInactiveConfigs();
+    let config = BoshDirectorClient.getPrimaryConfig(coreConfigs);
     super({
-      baseUrl: boshDirector.url,
-      auth: {
-        user: boshDirector.username,
-        pass: boshDirector.password
-      },
       headers: {
         Accept: 'application/json,text/plain;q=0.9'
       },
       followRedirect: false,
-      rejectUnauthorized: !boshDirector.skip_ssl_validation
+      rejectUnauthorized: !config.skip_ssl_validation
+    });
+    this.activeConfigs = activeConfigs;
+    this.coreConfigs = coreConfigs;
+    this.inactiveConfigs = inactiveConfigs;
+    this.config = config;
```

Comment:
This is not required. At this [location](https://github.com/muralisc/service-fabrik-broker/blob/7eb5a8ae011371271b99d0852c51e7cd4b69fc2a/lib/bosh/BoshDirectorClient.js#L262) you can just do this instead
```
 }, 302, _.sampe(this.activePrimary))
```
In fact its better to do this in createUpdateDeployment method.
```
createOrUpdateDeployment(manifest, opts) {
...
const config =  _.sampe(this.activePrimary);
if(config === undefined || config.length === 0) {
    throw new errors.NotFound('Did not find any bosh director config which supports creation of deployment');
}
...
```